src/pages/myAppointments.tsx

```diff
@@ -0,0 +1,58 @@
+import AppointmentCard from "@/components/appointment/AppointmentCard";
+import Navbar from "@/components/public/Navbar";
+import { GetAppointmentsByUserId } from "@/services/Appointment";
+import { Appointment } from "@/types/Appointment";
+import { Typography, Grid, Box } from "@mui/material";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { useContext, useEffect, useState } from "react";
+import { Database } from "supabase/db_types";
+import { userContext } from "supabase/user_context";
+import { NextRouter, useRouter } from "next/router";
+import { PagePaths } from "enum/pages";
+
+export default function Home() {
+    const router: NextRouter = useRouter();
+
+    const supabaseClient = useSupabaseClient<Database>();
+    const [appointments, setAppointments] = useState<Appointment[]>([]);
+    const userStatus = useContext(userContext);
+
+    function handleCardClick(appointmentId: string): void {
+        router.push(PagePaths.myAppointments + appointmentId);
```

Comment:
Seems like if I just change `appointmentId` in the URL path, I can access other appointment that I am not invited. Is there any chance you could fix this? This should not happen.

Reply:
done solve